src/resolve/resolved-dns-query.c

```diff
@@ -323,11 +343,32 @@ void dns_query_candidate_notify(DnsQueryCandidate *c) {
 
         state = dns_query_candidate_state(c);
 
-        if (DNS_TRANSACTION_IS_LIVE(state))
+        if (DNS_TRANSACTION_IS_LIVE(state)) {
+                DnsTransaction *t;
+                bool at_least_one_succeeded = false;
+                SET_FOREACH(t, c->transactions) {
+                        if (t->state == DNS_TRANSACTION_SUCCESS) {
+                                at_least_one_succeeded = true;
+                                break;
+                        }
+                }
```

Comment:
no {} around single line if blocks. i.e. remove the ones outer {} here

Reply:
yeepers, done